src/osd/ReplicatedBackend.cc

```diff
@@ -1316,6 +1316,12 @@ void ReplicatedBackend::calc_head_subsets(
   uint64_t size = obc->obs.oi.size;
   if (size)
     data_subset.insert(0, size);
+  map<hobject_t, pg_missing_t::item, hobject_t::ComparatorWithDefault>::const_iterator it
+    = missing.missing.find(head);
+  assert(it != missing.missing.end());
+  can_recover_partial = it->second.can_recover_partial;
+  if (can_recover_partial)
+    data_subset.intersection_of(it->second.dirty_extents);
```

Comment:
yeah， so I may use subtract_of(cloning) instead of subtract, which just subtract (the intersection of cloning and data_subset)